### DISCUSSION TITLE
Auto feature engineering / data pre-processing

### DISCUSSION THREAD

**Discussion by xavierXV3** [winedarksea/AutoTS#239] [Q&A]
Hello,

I read the doc and had a quick look at the code. In the code, I saw components related to feature engineering .
Is there a full documentation about auto feature engineering and data pre-processing?

Regards

**Comment by winedarksea** [comment]
The preprocessing is decently well documented. It is everything you see in the Transformation parameters. It can be used standalone separately from the rest of AutoTS.
The feature engineering is more model specific, not well documented. It mostly applies to all of the `~Regression` type models.

Calendar features are pretty well generated internally, but you can definitely pass one as a future_regressor easily enough.

**Reply by xavierXV3** [reply]
Thanks. I benchmarked AutoTS and many other tools. AutoTS outperforms them.

**Reply by winedarksea** [reply]
Out of curiosity, what was the second best framework you used?

**Reply by xavierXV3** [reply]
Without additional data preprocessing and feature engineering : AutogluonTS.
With them: Pycaret.
But they are somehow complementary as Pycaret includes state-of-the-art boosting whereas Autogluon includes soa deep learning.

**Reply by xavierXV3** [reply]
In fact with additional feature engineering and pre-processing I can currently obtain for some series very slightly better results with Pycaret. Maybe I could further tune AutoTS. I will have a look at it.
